### PR TITLE
Fuzz: random RLP generation

### PR DESCRIPTION
**Pull request by winsvega**
valid RLP generator

### CODE REVIEW

File: test/fuzzTesting/createRandomTest.cpp [hunk 1]
```diff
@@ -139,6 +141,14 @@ int main(int argc, char *argv[])
 			else
 				fillRandomTest(dev::test::doVMTests, (filltest) ? testFillString : c_testExampleVMTest, filldebug);
 		}
+		else
+		if (testSuite == "RLPTests")
+		{
+			if (checktest)
+				return checkRandomTest(dev::test::doStateTests, testmValue, debug);
```
Comment (CJentzsch) [inline test/fuzzTesting/createRandomTest.cpp]: I dont understand the logic her. You run StaeTests, although the testSuite = RLPTests?

File: test/fuzzTesting/fuzzHelper.cpp [hunk 2]
```diff
@@ -42,11 +42,121 @@ boostIntGenerator RandomCode::randOpLengGen = boostIntGenerator(gen, opLengDist)
...
+		int rnd = randUniIntGen() % 100;
+		if (rnd < 50)
+		{
+			//make header as array
+			if (length <= 55)
+				header = toCompactHex(128 + length);
+			else
+			{
+				std::string hexlength = toCompactHex(length);
+				header = toCompactHex(183 + hexlength.size()/2) + hexlength;
```
Comment (CJentzsch) [inline test/fuzzTesting/fuzzHelper.cpp]: space before and after /

File: test/fuzzTesting/fuzzHelper.cpp [hunk 3]
```diff
@@ -42,11 +42,121 @@ boostIntGenerator RandomCode::randOpLengGen = boostIntGenerator(gen, opLengDist)
...
+		}
+		else
+		{
+			//make header as list
+			if (length <= 55)
+				header = toCompactHex(192 + length);
+			else
+			{
+				std::string hexlength = toCompactHex(length, HexPrefix::DontAdd, 1);
+				header = toCompactHex(247 + hexlength.size()/2) + hexlength;
```
Comment (CJentzsch) [inline test/fuzzTesting/fuzzHelper.cpp]: space before and after /

File: test/fuzzTesting/fuzzHelper.cpp [hunk 4]
```diff
@@ -42,11 +42,121 @@ boostIntGenerator RandomCode::randOpLengGen = boostIntGenerator(gen, opLengDist)
...
+			}
+		}
+		result = header + result;
+		return result.size()/2;
```
Comment (CJentzsch) [inline test/fuzzTesting/fuzzHelper.cpp]: space before and after /

File: test/fuzzTesting/fuzzHelper.h [hunk 5]
```diff
@@ -83,6 +86,7 @@ class RandomCode
 	static std::string fillArguments(dev::eth::Instruction _opcode, RandomCodeOptions const& _options);
 	static std::string getPushCode(int _value);
 	static std::string getPushCode(std::string const& _hex);
+	static int recursiveRLP(std::string &result, int depth);
```
Comment (CJentzsch) [inline test/fuzzTesting/fuzzHelper.h]: _ before function parameter

File: test/fuzzTesting/createRandomTest.cpp (hunk 1 above)
Comment (winsvega) [inline test/fuzzTesting/createRandomTest.cpp]: flag:  in progress
I will update this PR when https://github.com/ethereum/cpp-ethereum/pull/2375 will be merged

File: test/fuzzTesting/fuzzHelper.cpp [hunk 6]
```diff
@@ -42,11 +42,153 @@ boostIntGenerator RandomCode::randOpLengGen = boostIntGenerator(gen, opLengDist)
...
+		_result = header + _result;
+		_debug = "[" + header + "(" + toString(length) + "){" + toString(rtype) + "}]" + _debug;
+		return _result.size() / 2;
+	}
+	if (_depth == 1)
+	{
+		bool genbug = false;
+		bool genbug2 = false;
+		int bugprobability = randUniIntGen() % 100;
+		if (bugprobability < 150 && !genvalidrlp)
```
Comment (gavofyork) [inline test/fuzzTesting/fuzzHelper.cpp]: camelCase

File: test/fuzzTesting/fuzzHelper.cpp [hunk 7]
```diff
@@ -42,11 +42,153 @@ boostIntGenerator RandomCode::randOpLengGen = boostIntGenerator(gen, opLengDist)
 boostIntGenerator RandomCode::randUniIntGen = boostIntGenerator(gen, uniIntDist);
 boostUInt64Generator RandomCode::randUInt64Gen = boostUInt64Generator(gen, uInt64Dist);
 
-std::string RandomCode::rndByteSequence(int _length, SizeStrictness _sizeType)
+int RandomCode::recursiveRLP(std::string &_result, int _depth, std::string &_debug)
+{
+	bool genvalidrlp = false;
```
Comment (gavofyork) [inline test/fuzzTesting/fuzzHelper.cpp]: camelCase

File: test/fuzzTesting/fuzzHelper.cpp [hunk 8]
```diff
@@ -42,11 +42,153 @@ boostIntGenerator RandomCode::randOpLengGen = boostIntGenerator(gen, opLengDist)
 boostIntGenerator RandomCode::randUniIntGen = boostIntGenerator(gen, uniIntDist);
 boostUInt64Generator RandomCode::randUInt64Gen = boostUInt64Generator(gen, uInt64Dist);
 
-std::string RandomCode::rndByteSequence(int _length, SizeStrictness _sizeType)
+int RandomCode::recursiveRLP(std::string &_result, int _depth, std::string &_debug)
```
Comment (gavofyork) [inline test/fuzzTesting/fuzzHelper.cpp]: `std::string& _debug`

**Comment by gavofyork** [comment]
a few style errors...

**Comment by subtly** [comment]
Linux compile fails.

File: test/fuzzTesting/fuzzHelper.h [hunk 9]
```diff
@@ -83,6 +99,7 @@ class RandomCode
 	static std::string fillArguments(dev::eth::Instruction _opcode, RandomCodeOptions const& _options);
 	static std::string getPushCode(int _value);
 	static std::string getPushCode(std::string const& _hex);
+	static int recursiveRLP(std::string &_result, int _depth, std::string& _debug);
```
Comment (arkpar) [inline test/fuzzTesting/fuzzHelper.h]: std::string& _result

**Pull request merged**